### ISSUE TITLE
Switch from `bytes` to `byteorder` crate

### ISSUE THREAD

**Issue by Snowiiii** [Snowiiii/Pumpkin#396]
### I've searched existing issues and couldn't find a duplicate.

- [X] I confirm this is not a duplicate.

### Is your feature request related to a problem? Please describe.

Currently we are using the very nice [bytes](https://github.com/tokio-rs/bytes) crate made by the tokio team. The crate is very nice and has all the features we need. 
The only problem is that it panics. This is very annoying and is a no-go in something like a Server. We often will read bytes coming from the Client, but we can't trust the client. It is also not easy to always check for remaining bytes ourself every time.
Currently we are using are own [ByteBuffer](https://github.com/Snowiiii/Pumpkin/blob/f587cb6748bae28443f19f9ecc80d1d24d0c3a1e/pumpkin-protocol/src/bytebuf/mod.rs#L12) struct for wrapping all read functions into checks. 

### Describe the solution you'd like

I would consider using [byteorder](https://github.com/BurntSushi/byteorder) instead of `bytes`. It returns Results instead of just a panic (which is how it should be in a library IMOP). 
This would make the codebase safer and prevent panics

**Comment by StripedMonkey** [comment]
I personally am a little mixed on this, because it introduces yet another type that boils down to a wrapper around `u8`s for what are incredibly tiny (code/implementation wise) reasons. I agree that we should avoid panics where possible, but we're ultimately never going to get rid of the `bytes` crate even if we do switch over to using byeorder/bytebuffer due to our dependency on `tokio` in the first place.

A compromise might be to introduce a blanket implementation of [a fallible check](https://github.com/tokio-rs/bytes/issues/254#issuecomment-1214137482) on Bufs of our own making. This would eliminate the need for a new buffer type, while giving us some guarantees about safety. To do this "correctly" we would have to figure out a way to enforce usage of those fallible versions though.

ByteBuffer itself is actually something I hope can eliminated in favor of a generic Serializer/Deserializer struct mind you, in my (now closed) PRs related to this, I put quite a bit of effort into trying to extract it and turn it into a Buf so I'm certainly aligned with that idea. It was a big annoyance for me that we had so many different types of buffers everywhere.

Please note that we are actually currently already including `byteorder 1.5.0` in our dependency tree, as `rsa`, `fastnbt`, and  `rand`/`zerocopy`  all depend on `byteorder`, so while it's not an exported dependency of these crates (as far as I know) it is still present as a dependency. Adding it wouldn't actually increase the number of crates we depend on.


tl;dr

I would prefer to have a blanket extension trait to add fallible conversions, and push for using those rather than adding *yet another* u8 wrapper since Pumpkin will *still* have to depend on bytes regardless. However, considering its usage in the ecosystem already in use, it's not the end of the world.

**Comment by Snowiiii** [comment]
> Next release of bytes will include try_get methods that do not panic: [tokio-rs/bytes#254](https://github.com/tokio-rs/bytes/issues/254)

Oh nice, great work @hpenne 👍
